applications/callflow/src/cf_remote_action.erl

```diff
@@ -0,0 +1,81 @@
+%%%-----------------------------------------------------------------------------
+%%% @copyright (C) 2011-2019, 2600Hz
+%%% @doc Executes remote callflow actions.
+%%%
+%%% This Source Code Form is subject to the terms of the Mozilla Public
+%%% License, v. 2.0. If a copy of the MPL was not distributed with this
+%%% file, You can obtain one at https://mozilla.org/MPL/2.0/.
+%%%
+%%% @end
+%%%-----------------------------------------------------------------------------
+-module(cf_remote_action).
+
+-export([handle/2
+        ,handle/3
+        ]).
+
+-include("callflow.hrl").
+
+-spec handle(kz_json:object(), kapps_call:call()) -> 'ok'.
+handle(Data, Call) ->
+    Action = kapps_call:kvs_fetch('remote-action', Call),
+    handle(Action, Data, Call).
+
+-spec handle(atom() | kz_term:ne_binary(), kz_json:object(), kapps_call:call()) -> 'ok'.
+handle(Action, Data, Call) ->
+    Q = kapps_call:controller_queue(Call),
+    ActionId = kz_binary:rand_hex(16),
+    API = [{<<"Call">>, call_to_jobj(Call)}
+          ,{<<"Data">>, Data}
+          ,{<<"Call-ID">>, kapps_call:call_id_direct(Call)}
+          ,{?KEY_MSG_ID, ActionId}
+           | kz_api:default_headers(Q, ?APP_NAME, ?APP_VERSION)
+          ],
+    kapi_callflow:publish_action_execute(kz_term:to_binary(Action), API),
+    wait_for_msg(ActionId, Call).
+
+-spec call_to_jobj(kapps_call:call()) -> kz_json:object().
+call_to_jobj(Call) ->
+    Flow = kapps_call:kvs_fetch('cf_flow', kz_json:new(), Call),
+    Keys = kz_json:get_keys(kz_json:get_json_value(<<"children">>, Flow, kz_json:new())),
+
+    Routines = [{fun kapps_call:kvs_erase/2, ['cf_flow']}
+               ,{fun kapps_call:kvs_store/3, 'action-exit-keys', Keys}
+               ,fun kapps_call:clear_helpers/1
+               ],
+    kapps_call:to_json(kapps_call:exec(Routines, Call)).
+
+-spec wait_for_msg(atom() | kz_term:ne_binary(), kapps_call:call()) -> 'ok'.
+wait_for_msg(ActionId, Call) ->
+    receive
```

Comment:
It seems like a "reasonable" timeout should be included here. Might also use `kapps_call_command:receive_event/1` instead of naked receive?

Reply:
`kapps_call_command:receive_event/1`  doesn't handle `{'amqp_msg', {'amqp_return', _JObj, _Returned}}`

Reply:
about the `timeout`, if there's no handler for the `action` and because of `mandatory`, we will receive the `amqp_return` message. if an application handles it, it needs to send back a message.